tests/integration/conftest.py

```diff
@@ -81,9 +82,48 @@ def assert_cairo_test_cases(
         skipped=set(expected_skipped_test_cases_names),
     )
 
+    name_to_test_case_result = {
+        test_result.test_case_name: test_result
+        for test_result in testing_summary.test_results
+        if isinstance(test_result, TestCaseResult)
+    }
+    diff = show_diff_between_cairo_test_cases(
+        name_to_test_case_result, expected, actual
+    )
+    if diff:
+        assert False, diff
     assert actual == expected
```

Comment:
isn't this redundant since we're asserting above?

Reply:
I am not sure if the diff assertion catches all edge cases. I left the second assertion as a "safety net".